cmd/check.go

```diff
@@ -25,4 +27,28 @@ func doCheck(ctx *cli.Context) error {
 	default:
 		return errors.Errorf("invalid storage type %v", config.StorageType)
 	}
+
+	return verifyNewUIDMap(ctx)
+}
+
+func verifyNewUIDMap(ctx *cli.Context) error {
+	binFile, err := exec.LookPath("newuidmap")
+	if err != nil {
+		return errors.Wrapf(err, "newuidmap not found in path")
+	}
+
+	fileInfo, err := os.Stat(binFile)
+	if err != nil {
+		return errors.Wrapf(err, "couldn't stat file: %s", binFile)
+	}
+
+	if fileInfo.Mode()&0100 == 0 {
+		return errors.Errorf("%s is not executable", binFile)
+	}
+
+	if fileInfo.Mode()&fs.ModeSetuid == 0 {
```

Comment:
There are distros where newuidmap is not setuid-root, but has the CAP_SYS_ADMIN and CAP_SETUID filecaps instead.  Here things become murky, though, because the file could just have the inheritable and effective caps, and the host give permitted users the inheritable process caps,   But I doubt anyone does that, so just checking for permitted and effective is probably best.